credentials/xds/xds_client_test.go

```diff
@@ -560,7 +562,10 @@ func (s) TestClientCredsProviderSwitch(t *testing.T) {
 	// Create a new root provider which uses the correct trust roots. And update
 	// the HandshakeInfo with the new provider.
 	root2 := makeRootProvider(t, "x509/server_ca_cert.pem")
-	handshakeInfo.SetRootCertProvider(root2)
+	handshakeInfo = xdsinternal.NewHandshakeInfo(root2, nil, []matcher.StringMatcher{matcher.StringMatcherForTesting(newStringP(defaultTestCertSAN), nil, nil, nil, nil, false)}, false)
+	uPtr = unsafe.Pointer(handshakeInfo)
+	addr = xdsinternal.SetHandshakeInfo(resolver.Address{}, &uPtr)
+	ctx = icredentials.NewClientHandshakeInfoContext(ctx, credentials.ClientHandshakeInfo{Attributes: addr.Attributes})
```

Comment:
Can we just update the existing pointer instead of creating a new entry in the context? That's how the production code will work.

Reply:
Ah, good point. Changed.